packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1828,33 +1911,79 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
         scrollbarPainter.update(metrics, metrics.axisDirection);
       }
     } else if (notification is ScrollEndNotification) {
-      if (_startDragScrollbarAxisOffset == null) {
+      if (_thumbDrag == null) {
         _maybeStartFadeoutTimer();
       }
     }
     return false;
   }
 
+  void _handleThumbDragDown(DragDownDetails details) {
+    handleThumbPress();
+  }
+
+  void _handleThumbDragStart(DragStartDetails details) {
+    handleThumbPressStart(details.localPosition);
+  }
+
+  void _handleThumbDragUpdate(DragUpdateDetails details) {
+    handleThumbPressUpdate(details.localPosition);
+  }
+
+  void _handleThumbDragEnd(DragEndDetails details) {
+    handleThumbPressEnd(details.localPosition, details.velocity);
+  }
+
+  void _handleThumbDragCancel() {
+    if (_gestureDetectorKey.currentContext == null) {
+      // The cancel was caused by the GestureDetector getting disposed, which
+      // means we will get disposed momentarily as well and shouldn't do
+      // any work.
+      return;
+    }
+    // _thumbHold might be null if the drag started.
+    // _thumbDrag might be null if the drag activity ended and called _disposeThumbDrag.
+    assert(_thumbHold == null || _thumbDrag == null);
+    _thumbHold?.cancel();
+    _thumbDrag?.cancel();
+    assert(_thumbHold == null);
+    assert(_thumbDrag == null);
+  }
+
+  void _initThumbDragGestureRecognizer(DragGestureRecognizer instance) {
+    instance.onDown = _handleThumbDragDown;
+    instance.onStart = _handleThumbDragStart;
+    instance.onUpdate = _handleThumbDragUpdate;
+    instance.onEnd = _handleThumbDragEnd;
+    instance.onCancel = _handleThumbDragCancel;
+    instance.gestureSettings = const DeviceGestureSettings(touchSlop: 0);
+    instance.dragStartBehavior = DragStartBehavior.down;
+  }
+
   Map<Type, GestureRecognizerFactory> get _gestures {
     final Map<Type, GestureRecognizerFactory> gestures = <Type, GestureRecognizerFactory>{};
-    if (_effectiveScrollController == null || !enableGestures) {
+    if (_axis == null || !enableGestures) {
       return gestures;
```

Comment:
If _axis is null, should gestures be cleared?

Reply:
Yes. If we don't (yet) know what the scrollbar's axis is then we ignore input gestures.

I restored this short-circuit to the way it was, just to reduce the scrollbar.dart diff count a little.